api/v1beta1/activemqartemis_types.go

```diff
@@ -370,6 +370,33 @@ type DeploymentPlanType struct {
 	// Specifies the Revision History Limit of the statefulset
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Revision History Limit"
 	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty"`
+	// Specifies the template for various resources that the operator controls
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Resource Templates"
+	Templates []TemplateType `json:"templates,omitempty"`
+}
+
+type TemplateType struct {
+	// Select which resources to match, an empty selector will match all resources
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Selector"
+	Selector *ResourceMatch `json:"selector,omitempty"`
+
+	// Custom annotations
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Annotations"
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Custom labels
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Labels"
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// match criteria to restrict the selection of resources
+type ResourceMatch struct {
+	// APIGroup is the group version string of resources to match
+	APIGroup *string `json:"apiGroup,omitempty"`
+	// Kind is the type of resources to match
+	Kind *string `json:"kind,omitempty"`
+	// Name is the pattern matcher for the Name of resources to match
+	Name *string `json:"name,omitempty"`
 }
```

Comment:
[OPTIONAL] looking the kubernetes api, I found something similar fileds/structs in the [StatefulSetSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#statefulsetspec-v1-apps)
```
selector
LabelSelector
...
template
PodTemplateSpec
...
```

If my understanding is correct (BIG IF 😅) `TemplateType` could also be `ResourceTemplateSpec` and `ResourceMatch` could also be `ResourceSelector`, WDYT?

Reply:
for sure Type does not bring much value, I was blindly following AddressType/AcceptorType etc... let me revisit. Thanks!

Reply:
same holds for the Spec postfix, am thinking TemplateType could also be ResourceTemplate